.github/workflows/deploy_aws_api_infrastructure_dev.yml

```diff
@@ -1,17 +1,93 @@
-name: Deploy AWS API Infrastructure - Dev
-
-# NOTE: Not yet functional - hopefully will be shortly :)
+name: Deploy AWS API Infrastructure
 
 on:
   push:
+    branch:
+      - main
+    paths:
+      - 'api/awsauth/**'
 
+  workflow_dispatch:
+    deploy_to_prod:
+      description: 'If "true" will run full CI pipeline and deploy to prod'
+      default: "false"
 
 defaults:
   run:
     working-directory: api/awsauth/
 
 jobs:
-  deploy-dev:
+
+  DeployDev:
+    environment:
+      name: dev
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install Serverless Framework and NPM dependencies
+      run: |
+        npm install -g serverless
+        npm install
+    - name: Deploy
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        CLOUDFRONT_DISTRIBUTION_ID: ${{ secrets.CLOUDFRONT_DISTRIBUTION_ID }}
+        DOTENV: ${{ secrets.DOTENV }}
+        AWS_REGION: us-east-1
+      run: tools/deploy-api-infrastructure.sh dev
+
+  TestDev:
+    if: github.event.inputs.deploy_to_prod == 'true'
+    needs: [DeployDev]
```

Comment:
FWIW- If we just always ran TestDev, you could combine TestDev and DeployDev jobs and speed up the github actions runtime (avoid checking out / installing python redundantly, etc.).